### PR TITLE
feat(phmfactory): verbalize interpretable traces through PHM-EIR

### PR DESCRIPTION
**Pull request by liq22**
## Purpose

Add a minimal, provider-neutral path for using an LLM to explain PHMFactory's intrinsically interpretable models without asking the LLM to infer a mechanism from raw tensors.

The supported flow is:

```text
interpretable PHM model
→ same-forward model-native trace
→ PHM Explanation Intermediate Representation (PHM-EIR)
→ explicit user-supplied LLM callback
→ validated public explanation
```

## Public contract

Adds `phmfactory.explanation` with:

- `PHMExplanationState` (`phm-eir/v1`): prediction, typed evidence atoms, evidence paths, same-forward class contributions, explicitly supplied mechanism relations, uncertainty, operating conditions, capabilities, and limitations;
- explicit adapters for `XOANOperatorPath.forward_evidence` and `TSPN_UXFD.forward_with_fuzzy_trace`;
- `build_llm_packet`, `explain_with_llm`, and `parse_llm_explanation`;
- deterministic rejection of unknown evidence, path, or relation identifiers.

The package contains no provider SDK, API-key handling, retry loop, provider substitution, parser repair, or hidden fallback. The caller supplies exactly one `generate(packet)` function.

## Scientific boundary

- A model-native operator path, fuzzy rule, feature, or attention weight is not labeled a physical mechanism unless an independent mapping explicitly supplies that relation.
- Plain TSPN is not certified for case-level evidence verbalization because its normal forward returns logits without a same-forward trace.
- Structured-output and identifier checks establish referential closure only; they do not prove semantic or mechanism faithfulness.
- The engineering layer is intended to support later intervention-based evaluation, not to create a positive research result by construction.

## Tests

Adds focused tests for:

- schema references and unknown-ID rejection;
- XOAN operator-path adaptation;
- TSPN-UXFD fuzzy contribution reconstruction;
- provider-neutral packet construction;
- one-call LLM callback behavior with no fallback.

Local focused result before push:

```text
6 passed
```

The branch starts directly from current `dev` and is not behind it.

### CODE REVIEW

**Comment by liq22** [comment]
## Paper-split boundary

This runtime PR is now assigned to the **TII industrial paper** only.

Its scientific role is bounded to:

```text
same-forward interpretable PHM trace
→ typed PHM explanation state
→ query-conditioned industrial representation
→ provider-neutral LLM verbalization
→ deterministic reference validation
```

The TII paper will evaluate XOAN operator paths and TSPN-UXFD fuzzy traces under executable operator/rule interventions and a calibrated speed change. The runtime PR must not claim a universal natural-language explanation theory, recovery of hidden reasoning, or physical validity of learned operators/rules.

The separate TPAMI track owns the general results on interventional semantic bisimulation, coarsest dynamic representations, collision lower bounds, information limits, and multi-domain evaluation. Those results should not be implemented or advertised as part of this PHMFactory PR.

Engineering implication for this PR:

- keep PHM-EIR/provider-neutral callback and explicit adapters;
- add query-closed PHM Native Explanation Graph support only when its deterministic closure and reconstruction tests are specified;
- retain full native state as the safe baseline;
- do not add a learned generic bottleneck or multi-domain abstraction here;
- keep invalid references, omissions, and unsupported mechanism mappings visible rather than repaired.

The two-paper architecture is tracked in `AI4Engineering-L/P03-Evidence-Grounded-LLM-XFD` PR #21.

File: phmfactory/explanation/adapters.py [hunk 1]
```diff
@@ -0,0 +1,523 @@
...
+            ]
+            + (["decision_reconstruction"] if len(ranked) == len(firing) else ["partial_contribution"])
+        ),
+        limitations=tuple(limitations),
+        metadata=freeze_mapping(
+            {
+                "fuzzy_scale": scale,
+                "exported_rule_count": len(ranked),
+                "total_rule_count": len(firing),
+            },
```
[the remark this answers is not in the document]
Comment (liq22) [reply phmfactory/explanation/adapters.py:528]: Fixed in 1e25caef. The exported metadata now contains the selected sample's effective rule_mask and consequent_permutation from the same forward trace. Both shared [rules] and per-sample [batch,rules] native permutations are handled; missing external trace values are explicitly disclosed as unknown. Added real-model tests for both shapes and comparison with the unmodified trace. Final-head CI is running; no model implementation or training semantics changed.

File: phmfactory/explanation/adapters.py [hunk 2]
```diff
@@ -0,0 +1,523 @@
...
+    total = sum(exponentials)
+    return tuple(value / total for value in exponentials)
+
+
+def _labels(class_names: Sequence[str] | None, size: int) -> tuple[str, ...]:
+    if class_names is None:
+        return tuple(f"class_{index}" for index in range(size))
+    labels = tuple(str(item).strip() for item in class_names)
+    if len(labels) != size or any(not item for item in labels):
+        raise ValueError(f"class_names must contain exactly {size} non-empty labels")
```
[the remark this answers is not in the document]
Comment (liq22) [reply phmfactory/explanation/adapters.py:108]: Fixed in 1e25caef. _labels now rejects duplicate normalized class names before prediction or contribution records are created. The added real TSPN-UXFD tests use an explicit two-class model and cover both exact duplicates and duplicates after trimming. No automatic renaming or class-index remapping is introduced.

File: phmfactory/explanation/adapters.py [hunk 3]
```diff
@@ -0,0 +1,491 @@
...
+    )
+
+    evidence_atoms: list[EvidenceAtom] = [path_atom]
+    metrics: list[tuple[str, float]] = []
+    for key, display_name in (
+        ("logit_relative_rmse", "relaxed_discrete_logit_relative_rmse"),
+        ("predictive_entropy", "predictive_entropy"),
+        ("dictionary_insufficiency_score", "dictionary_insufficiency_score"),
+        ("relative_rmse", "relaxed_discrete_signal_relative_rmse"),
+        ("normalized_selection_entropy", "normalized_selection_entropy"),
```
[the remark this answers is not in the document]
Comment (liq22) [reply phmfactory/explanation/adapters.py]: The final adapter reads normalized_sparsemax_selection_entropy, matching the native XOAN report. The selected inference route now also controls the prediction and predictive entropy, and real-model tests cover both relaxed and discrete modes. Final candidate 7556b1e2 passed all four current-head workflows.

**Review by liq22 — COMMENTED** [review]
Final bounded review under the maintainer's explicit authorization:

- Existing XOAN and TSPN-UXFD model implementations are reused, not replaced or imported wholesale from obsolete branches.
- The adapter preserves the selected relaxed/discrete inference route, reads the native entropy field, exports same-forward fuzzy contributions, and does not claim full reconstruction when rules are truncated.
- Duplicate normalized class names now fail before target construction; actual per-sample rule masks and shared/per-sample consequent permutations are included. Missing intervention values are explicitly unknown.
- LLM callback remains explicit and single-call; no provider SDK, response repair, retry, substitute model, or hidden physical-mechanism claim.
- Both explanation test files are in the existing runtime test selection. Final-head Core quality gates and public-package workflows passed, including real-model and Dummy paths. Repository layout and submodule policy also passed.
- All three inline review threads were answered and resolved. Changelog is recorded at doc/changelog/2026-09-07-explanation-integration.md.

This supports merging the bounded engineering interface, not a claim of semantic/mechanistic explanation faithfulness or a new benchmark result. No THU rerun or release-status modification.
